### ISSUE TITLE
Unable to get network stats from pid

### ISSUE THREAD

**Issue by gl328518397** [google/cadvisor#1522]
my OS is redhat7,
docker is 1.8
my cmd is 
"
 docker run --volume=/:/rootfs:ro  \
--volume=/var/run:/var/run:rw \
--volume=/sys:/sys:ro \
--volume=/var/lib/docker/:/var/lib/docker:ro \
--volume=/cgroup:/cgroup:ro  \
--publish=8096:8080 \
--detach=true \
--link influxsrv:influxsrv \
--privileged=true \
 google/cadvisor:latest  \
  -v=2 \
 -storage_driver=influxdb \
 -storage_driver_db=cadvisor \
 -storage_driver_host=127.0.0.1:8086 
"

i say logs -->>
Unable to get network stats from pid 34215: couldn't read network stats: failure opening /rootfs/proc/34215/net/dev: open /rootfs/proc/34215/net/dev: no such file or directory
I1101 06:01:03.346357       1 helpers.go:101] Unable to get network stats from pid 32943: couldn't read network stats: failure opening /rootfs/proc/32943/net/dev: open /rootfs/proc/32943/net/dev: no such file or directory
I1101 06:01:04.145622       1 helpers.go:101] Unable to get network stats from pid 34215: couldn't read network stats: failure opening /rootfs/proc/34215/net/dev: open /rootfs/proc/34215/net/dev: no such file or directory
I1101 06:01:05.720426       1 helpers.go:101] Unable to get network stats from pid 32943: couldn't read network stats: failure opening /rootfs/proc/32943/net/dev: open /rootfs/proc/32943/net/dev: no such file or directory
I1101 06:01:07.009749       1 helpers.go:101] Unable to get network stats from pid 32943: couldn't read network stats: failure opening /rootfs/proc/32943/net/dev: open /rootfs/proc/32943/net/dev: no such file or directory
I1101 06:01:07.710892       1 helpers.go:101] Unable to get network stats from pid 34215: couldn't read network stats: failure opening /rootfs/proc/34215/net/dev: open /rootfs/proc/34215/net/dev: no such file or directory
I1101 06:01:09.289906       1 helpers.go:101] Unable to get network stats from pid 34215: couldn't read network stats: failure opening /rootfs/proc/34215/net/dev: open /rootfs/proc/34215/net/dev: no such file or directory
I1101 06:01:10.660529       1 helpers.go:101] Unable to get network stats from pid 32943: couldn't read network stats: failure opening /rootfs/proc/32943/net/dev: open /rootfs/proc/32943/net/dev: no such file or directory

**Comment by dashpole** [comment]
what cadvisor version are you running?

**Comment by Dennisparchkov** [comment]
I'm getting the same issue as described:

I am am running latest cadvisor in a container in Centos7.

When i restart one of the containers i am monitoring using `docker restart ...` i get the following in the cadvisor logs:
```
 Unable to get network stats from pid 18927: couldn't read network stats: failure opening /│            - /var/lib/docker/:/var/lib/docker:ro
rootfs/proc/18927/net/dev: open /rootfs/proc/18927/net/dev: no such file or directory
```
Result is no Network stats but `has_network` is still true.

**Comment by dashpole** [comment]
Is the issue temporary or permanent?

**Comment by Dennisparchkov** [comment]
permanent issue, every monitoring container that i restart has this issue

**Comment by dashpole** [comment]
Yeah, this looks like a bug with the interaction of cAdvisor with `docker restart`.  cAdvisor assumes the container process stays the same for the lifetime of the container.  I ran a quick test, and `docker restart` keeps the container id the same, but changes the PID of the container.

There are a few things we could do:

* Recommend disabling network metrics when using docker's restart mechanism
* Periodically "refresh" the spec of the container from the docker daemon

Does the PID change only when `docker restart` is used, or can it change for other reasons (e.g. docker container exits, but a restart policy was specified)?

**Comment by Dennisparchkov** [comment]
Actions like starting and stopping containers `docker stop` and `docker start` also change its PID.

I was thinking of using the event mechanism from the go docker library used  (https://godoc.org/github.com/docker/docker/client#Client.Events) for monitoring these types of events that will change the PID and act accordingly. 

For this specific case when a `restart` event is triggered the PID of the container should be updated.

**Comment by dashpole** [comment]
That might be a better solution than a periodic refresh.  I would be curious to see how complex that is to implement.

**Comment by dims** [comment]
Closing this issue as it has been open since before January 1, 2024. If this is still relevant, please feel free to reopen or create a new issue.
